src/test/java/com/jakewharton/rxrelay2/BehaviorRelayTest.java

```diff
@@ -0,0 +1,341 @@
+        assertFalse(source.hasObservers());
+    }
+
+    @Test
+    public void testEmissionSubscriptionRace() throws Exception {
+        Scheduler s = Schedulers.io();
+        Scheduler.Worker worker = Schedulers.io().createWorker();
+        try {
+            for (int i = 0; i < 50000; i++) {
+                if (i % 1000 == 0) {
+                    System.out.println(i);
+                }
+                final BehaviorRelay<Object> rs = BehaviorRelay.create();
+
+                final CountDownLatch finish = new CountDownLatch(1);
+                final CountDownLatch start = new CountDownLatch(1);
+
+                worker.schedule(new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            start.await();
+                        } catch (Exception e1) {
+                            e1.printStackTrace();
+                        }
+                        rs.accept(1);
+                    }
+                });
+
+                final AtomicReference<Object> o = new AtomicReference<Object>();
+
+                rs.subscribeOn(s).observeOn(Schedulers.io())
+                .subscribe(new DefaultObserver<Object>() {
+
+                    @Override
+                    public void onComplete() {
+                        o.set(-1);
+                        finish.countDown();
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        o.set(e);
+                        finish.countDown();
+                    }
+
+                    @Override
+                    public void onNext(Object t) {
+                        o.set(t);
+                        finish.countDown();
+                    }
+
+                });
+                start.countDown();
+
+                if (!finish.await(5, TimeUnit.SECONDS)) {
+                    System.out.println(o.get());
+                    System.out.println(rs.hasObservers());
+                    fail("Timeout @ " + i);
+                    break;
+                } else {
+                    Assert.assertEquals(1, o.get());
+                }
+            }
+        } finally {
+            worker.dispose();
+        }
+    }
+
+    @Test
+    public void testCurrentStateMethodsNormalEmptyStart() {
+        BehaviorRelay<Object> as = BehaviorRelay.create();
+
+        assertFalse(as.hasValue());
+        assertNull(as.getValue());
+
+        as.accept(1);
+
+        assertTrue(as.hasValue());
+        assertEquals(1, as.getValue());
+    }
+
+    @Test
+    public void testCurrentStateMethodsNormalSomeStart() {
+        BehaviorRelay<Object> as = BehaviorRelay.createDefault((Object)1);
+
+        assertTrue(as.hasValue());
+        assertEquals(1, as.getValue());
+
+        as.accept(2);
+
+        assertTrue(as.hasValue());
+        assertEquals(2, as.getValue());
+    }
+
+    @Test
+    public void onNextNull() {
```

Comment:
If you want to make the same change here that'd be great. The tests are ignored for now though because of the memory issue and I probably won't be working on this for a week.